src/tests/illumidesk/authenticators/test_lti11_authenticator.py

```diff
@@ -298,6 +306,63 @@ async def test_authenticator_returns_auth_state_with_empty_lis_outcome_service_u
                 'course_id': 'intro101',
                 'lms_user_id': '185d6c59731a553009ca9b59ca3a885100000',
                 'user_role': 'Learner',
+                'workspace_type': 'notebook',
+            },
+        }
+        assert result == expected
+
+
+@pytest.mark.asyncio
+@patch('illumidesk.authenticators.authenticator.LTI11LaunchValidator')
+async def test_authenticator_returns_default_workspace_type_when_missing(lti11_authenticator,):
+    """
+    Do we get the default workspace_type when its not sent with the launch request?
+    """
+    with patch.object(LTI11LaunchValidator, 'validate_launch_request', return_value=True):
+        authenticator = LTI11Authenticator()
+        args = factory_lti11_complete_launch_args('canvas', 'Learner', 'notebook')
+        args['lis_outcome_service_url'] = [b'']
+        handler = Mock(
+            spec=RequestHandler,
+            get_secure_cookie=Mock(return_value=json.dumps(['key', 'secret'])),
+            request=Mock(arguments=args, headers={}, items=[],),
+        )
+        result = await authenticator.authenticate(handler, None)
+        expected = {
+            'name': 'student1',
+            'auth_state': {
+                'course_id': 'intro101',
+                'lms_user_id': '185d6c59731a553009ca9b59ca3a885100000',
+                'user_role': 'Learner',
+                'workspace_type': 'notebook',
+            },
+        }
+        assert result == expected
+
+
+@pytest.mark.asyncio
+@patch('illumidesk.authenticators.authenticator.LTI11LaunchValidator')
+async def test_authenticator_returns_custom_workspace_type_when_set(lti11_validator,):
+    """
+    Do we get the custom workspace_type when its sent with the launch request?
+    """
+    with patch.object(lti11_validator, 'validate_launch_request', return_value=True):
+        authenticator = LTI11Authenticator()
+        args = factory_lti11_complete_launch_args('canvas', 'Learner', 'foo')
+        args['lis_outcome_service_url'] = [b'']
+        handler = Mock(
+            spec=RequestHandler,
+            get_secure_cookie=Mock(return_value=json.dumps(['key', 'secret'])),
+            request=Mock(arguments=args, headers={}, items=[],),
+        )
+        result = await authenticator.authenticate(handler, None)
+        expected = {
+            'name': 'student1',
+            'auth_state': {
+                'course_id': 'intro101',
+                'lms_user_id': '185d6c59731a553009ca9b59ca3a885100000',
+                'user_role': 'Learner',
+                'workspace_type': 'foo',
```

Comment:
I think the result should be replaced with a valid and supported workspace from here to control our flow later in the code